circuit_knitting/cutting/qpd/decompose.py

```diff
@@ -198,6 +204,13 @@ def _decompose_qpd_instructions(
         data_id_offset += 1
         circuit.data.insert(i + data_id_offset, inst2)
 
+    # Get equivalence library
+    if basis_gate_set is not None:
+        basis_gate_set = basis_gate_set.lower()
+    else:
+        basis_gate_set = "standard"
+    equivalence = equivalence_libraries[basis_gate_set]
```

Comment:
If you do this, you won't need `equivalence_library` to be a `defaultdict`.  An ordinary `dict` will suffice.

Also, because currently `equivalence_library` is a `defaultdict`, this in theory means reading from it is not a pure/immutable operation, because the dict itself will be modified to add a new entry for the missing name (this is what `lambda: None` does).

```suggestion
    equivalence = equivalence_libraries.get(basis_gate_set)
```